plutus-core/plutus-ir/test/datatypes/maybe.golden

```diff
@@ -1,65 +1,22 @@
 (program 1.0.0
   [
-    [
-      [
-        (lam
-          Nothing_i0
-          (all a_i0 (type) [(lam a_i0 (type) (all out_Maybe_i0 (type) (fun out_Maybe_i1 (fun (fun a_i2 out_Maybe_i1) out_Maybe_i1)))) a_i1])
-          (lam
-            Just_i0
-            (all a_i0 (type) (fun a_i1 [(lam a_i0 (type) (all out_Maybe_i0 (type) (fun out_Maybe_i1 (fun (fun a_i2 out_Maybe_i1) out_Maybe_i1)))) a_i1]))
-            (lam
-              match_Maybe_i0
-              (all a_i0 (type) (fun [(lam a_i0 (type) (all out_Maybe_i0 (type) (fun out_Maybe_i1 (fun (fun a_i2 out_Maybe_i1) out_Maybe_i1)))) a_i1] (all out_Maybe_i0 (type) (fun out_Maybe_i1 (fun (fun a_i2 out_Maybe_i1) out_Maybe_i1)))))
-              [
-                { Just_i2 (all a_i0 (type) (fun a_i1 a_i1)) }
-                (abs a_i0 (type) (lam x_i0 a_i2 x_i1))
-              ]
-            )
-          )
-        )
-        (abs
-          a_i0
-          (type)
-          (abs
-            out_Maybe_i0
-            (type)
-            (lam
-              case_Nothing_i0
-              out_Maybe_i2
-              (lam case_Just_i0 (fun a_i4 out_Maybe_i3) case_Nothing_i2)
-            )
-          )
-        )
-      ]
+    (lam
```

Comment:
another missed reduction (I guess we're just stopping too soon indeed)

Reply:
There seems to be a renaming bug that we hit after 4 iterations of simplifier, so we probably need to address that first.